packages/blocks/src/__internal__/rich-text/rich-text-operations.ts

```diff
@@ -542,6 +543,32 @@ export function handleLineStartBackspace(page: Page, model: ExtendedModel) {
   handleUnknownBlockBackspace(model);
 }
 
+function handleParagraphBlockLeftKey(page: Page, model: ExtendedModel) {
+  if (!matchFlavours(model, ['affine:paragraph'])) return;
+  const pageElement = getDefaultPage(page);
+  if (!pageElement) {
+    // Maybe in edgeless mode
+    return;
+  }
+  const titleVEditor = pageElement.titleVEditor;
+  const parent = page.getParent(model);
+  if (parent && matchFlavours(parent, ['affine:frame'])) {
+    const frameParent = page.getParent(parent);
+    if (frameParent && matchFlavours(frameParent, ['affine:page'])) {
+      const frameIndex = frameParent.children.indexOf(parent);
```

Comment:
The page block may contains other blocks such as `affine:surface`, you can filter the children before check the index of frame, for example:
```ts
frameParent.children.filter(block => matchFlavours(block, ['affine:frame'])).indexOf(parent)
```